### PR TITLE
Product Gallery: disable prev/next buttons on first and last images

### LINKED ISSUE #53346: Product Gallery: disable prev/next buttons when on first or last image
Currently buttons are always active even though they do nothing when on first or last image.

We plan to implement the infinite scroll (https://github.com/woocommerce/woocommerce/issues/43255) but I'm not sure if can land in WooCommerce 9.6 hence I'd implement the disabling of those buttons.

### PR DESCRIPTION
**Pull request by kmanijak**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Till now the arrow buttons in Product Gallery are always active even on the first and last image when they do nothing.

We plan to implement the infinite scroll (https://github.com/woocommerce/woocommerce/issues/43255) in Product Gallery but I'm sure it won't land in WooCommerce 9.6 hence I decided to implement the temporary improvement which is disabling arrows on first and last images which feels like basic feature.

Closes https://github.com/woocommerce/woocommerce/issues/53346

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Make sure you have a product that has at least 4 images
1. Go to Editor > Single Product Template.
2. Replace Product Image Gallery with Product Gallery (Beta) block
3. Save and go to frontend to product that has at least 4 images
4. **Expected:**
    - The PREV arrow is disabled (because it's the first image)
    - The NEXT arrow is enabled
6. Move to the last image
7. **Expected:**
    - The PREV arrow gets enabled on second image
    - The NEXT arrow is disabled (because it's the last image in this view)
8. Click on the image to open full screen pop-up
9. **Expected:**
    - Both arrows are enabled (there's additional image available in this view)
10. Switch to last image
11. **Expected:**
    - The PREV arrow gets enabled on second image
    - The NEXT arrow is disabled (because it's the last image in this view)
12. Close the full screen pop-up
13. **Expected:**
    - First image is displayed in the on-page gallery (because the previous image is not available in this view)
    - The PREV arrow is disabled
    - The NEXT arrow is enabled
14. Switch to second image
15. Click on the image to open full screen pop-up
16. Close the full screen pop-up
13. **Expected:**
    - You stay on the same image, because it's available in both views

Before - Left arrow is active but does nothing when being on first image | After - left arrow is disabled on the first image
-- | --
image | image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductGalleryLargeImageNextPrevious.php [hunk 1]
```diff
@@ -213,7 +222,10 @@ protected function get_outside_button( $button_type, $context ) {
 		}
 
 		return sprintf(
-			'<button class="wc-block-product-gallery-large-image-next-previous--button wc-block-product-gallery-large-image-next-previous-%1$s--%2$s">
+			'<button
+				data-wc-bind--disabled="state.disable%2$s"
+				class="wc-block-product-gallery-large-image-next-previous--button wc-block-product-gallery-large-image-next-previous-%1$s--%3$s"
```
Comment (Aljullu) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductGalleryLargeImageNextPrevious.php]: If I'm reading it right, we are printing variable `%2$s` before `%1$s`, right? Is there any reason for that, I find it a bit confusing. :sweat_smile:

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductGalleryLargeImageNextPrevious.php [hunk 2]
```diff
@@ -172,11 +177,14 @@ protected function get_inside_button( $button_type, $context ) {
 		}
 
 		return sprintf(
-			'<button class="wc-block-product-gallery-large-image-next-previous--button wc-block-product-gallery-large-image-next-previous-%1$s--%2$s">
-				<svg  xmlns="http://www.w3.org/2000/svg" width="49" height="48" viewBox="0 0 49 48" fill="none">
+			'<button
+				data-wc-bind--disabled="state.disable%2$s"
+				class="wc-block-product-gallery-large-image-next-previous--button wc-block-product-gallery-large-image-next-previous-%1$s--%3$s"
```
Comment (Aljullu) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductGalleryLargeImageNextPrevious.php]: Same here.

**Review by Aljullu — APPROVED** [review]
I tested the different flows and everything is working as expected. Code changes look good to me as well. I only left a couple of minor questions inline. :point_down:

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductGalleryLargeImageNextPrevious.php (hunk 1 above)
Comment (kmanijak) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductGalleryLargeImageNextPrevious.php]: Good catch! Fixed in https://github.com/woocommerce/woocommerce/pull/53318/commits/5ee815f48f687bb214ae84cc899a9dfb4925b51e!

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductGalleryLargeImageNextPrevious.php (hunk 2 above)
Comment (kmanijak) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductGalleryLargeImageNextPrevious.php]: Same here.

**Pull request merged**